test-framework/security-oidc/src/main/java/io/quarkus/test/security/oidc/OidcTestSecurityIdentityAugmentorProducer.java

```diff
@@ -1,46 +1,64 @@
 package io.quarkus.test.security.oidc;
 
 import java.util.Map;
+import java.util.Optional;
 import java.util.stream.Collectors;
 
 import javax.enterprise.context.ApplicationScoped;
 import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+import javax.json.Json;
+import javax.json.JsonObjectBuilder;
 
+import org.eclipse.microprofile.config.inject.ConfigProperty;
 import org.eclipse.microprofile.jwt.Claims;
 import org.eclipse.microprofile.jwt.JsonWebToken;
 import org.jose4j.jwt.JwtClaims;
 
 import io.quarkus.arc.Unremovable;
 import io.quarkus.oidc.AccessTokenCredential;
 import io.quarkus.oidc.IdTokenCredential;
+import io.quarkus.oidc.OidcConfigurationMetadata;
+import io.quarkus.oidc.UserInfo;
 import io.quarkus.oidc.runtime.OidcJwtCallerPrincipal;
+import io.quarkus.oidc.runtime.OidcUtils;
 import io.quarkus.security.identity.SecurityIdentity;
 import io.quarkus.security.runtime.QuarkusSecurityIdentity;
 import io.quarkus.test.security.TestSecurityIdentityAugmentor;
 import io.smallrye.jwt.build.Jwt;
 import io.smallrye.jwt.util.KeyUtils;
+import io.vertx.core.json.JsonObject;
 
 @ApplicationScoped
 public class OidcTestSecurityIdentityAugmentorProducer {
 
+    @Inject
+    @ConfigProperty(name = "quarkus.oidc.token.issuer")
+    Optional<String> issuer;
+
+    private final static String CLAIM_PREFIX = "claim.";
```

Comment:
Can we do this in a type safe manner? I think it would be nicer to  add a specific annotation rather than having magic prefixes.

Reply:
Hi Stuart @stuartwdouglas, `claim.` prefix is not a new thing, it is already in `alpha1`/`alpha2`.

`TestSecurity` has already `SecurityAttribute`. 
Introducing `Claim`/`UserInfo`/`ConfigMetadata` attributes at the `TestSecurity` level would not really work well - as it is OIDC/JWT specific. Also `Claim` would make sense for testing `quarkus-smallrye-jwt`/`quarkus-oidc`, but `UserInfp/ConfigMetadata` - only `quarkus-oidc`. Hence I decided to go with this convention - it is documented.

So instead of:
```
@Test
    @TestSecurity(user = "userOidc", roles = "viewer", attributes = {
            @SecurityAttribute(key = "claim.email", value = "user@gmail.com")
    })
```

would you prefer:

```
@Test
    @TestSecurity(user = "userOidc", roles = "viewer", claims = {
            @Claim(key = "email", value = "user@gmail.com")
    })
```
 
? This is certainly cleaner but like I said, `Claim` has no meaning when testing for ex basic authentication.

Or we can do:
```
// quarkus-smallrye-jwt/quarkus-oidc
@Test
    @TestJwtSecurity(user = "userOidc", roles = "viewer", claims = {
            @Claim(key = "email", value = "user@gmail.com"),
    })

// quarkus-oidc only
@Test
    @TestOidcSecurity(user = "userOidc", roles = "viewer", claims = {
            @Claim(key = "email", value = "user@gmail.com"),
            @UserInfo(key = "sub", value = "user@gmail.com")
    })

```

This is probably the best - but we'd need to duplicate `test-framework/security` which takes care of `TestSecurity` to do the same for `TestJwtSecurity` and `TestOidcSecurity`. 

What do you think ?

Reply:
@stuartwdouglas I'll open a parallel draft PR with `security-common`, lets see how it will look

Reply:
Thinking about it a bit more I like the first one best, and just thrown an error if an invalid attribute is present (e.g. the user adds a claim but OIDC is not present).

OIDC is our preferred auth mechanism, I think it is fine to include OIDC properties here, even if they are not always relevant.

Reply:
@stuartwdouglas Makes sense, I've closed #17120 but will try to tweak this PR to support your extra annotation's idea from #17120